idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/typeclasses/TypeResolutionTest.kt

```diff
@@ -0,0 +1,96 @@
+package arrow.meta.ide.plugins.typeclasses
+
+import arrow.meta.ide.phases.config.buildFolders
+import arrow.meta.ide.testing.env.IdeHeavyTestSetUp
+import arrow.meta.plugin.testing.Assert
+import arrow.meta.plugin.testing.CompilerPlugin
+import arrow.meta.plugin.testing.CompilerTest
+import arrow.meta.plugin.testing.Dependency
+import arrow.meta.plugin.testing.Source
+import arrow.meta.plugin.testing.assertThis
+import com.intellij.openapi.application.WriteAction
+import com.intellij.openapi.roots.ModuleRootManager
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.SyntaxTraverser
+import com.intellij.testFramework.PsiTestUtil
+import org.jetbrains.kotlin.idea.KotlinFileType
+import org.jetbrains.kotlin.nj2k.postProcessing.type
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.types.ErrorType
+
+class TypeResolutionTest : IdeHeavyTestSetUp() {
+    fun testSetup() {
+        val codeSnippet = TypeclassesTestCode.c1
+        val module = myModule
+
+        // setup content-root
+        val roots = ModuleRootManager.getInstance(module).contentRoots
+        assertSize(1, roots)
+        val contentRoot = roots[0]
+
+        // then add and exclude content-root/build
+        val buildFolder = WriteAction.computeAndWait<VirtualFile, Throwable> {
+            contentRoot.createChildDirectory(this, "build")
+        }
+        PsiTestUtil.addExcludedRoot(module, buildFolder)
+
+        // then add content-root/src as source folder
+        val srcFolder = WriteAction.computeAndWait<VirtualFile, Throwable> {
+            contentRoot.createChildDirectory(this, "src")
+        }
+        PsiTestUtil.addExcludedRoot(module, srcFolder)
+
+        val buildFolders = project.buildFolders()
+        assertSize(1, buildFolders)
+
+        // copy code snippet into a new virtual file
+        // TODO: alternatively we could store a bunch of .kt files on disk and run the test for each
+        val sourceFile = WriteAction.computeAndWait<VirtualFile, Throwable> {
+            val sourceFile = srcFolder.createChildData(this, "source.kt")
+            sourceFile.setBinaryContent(codeSnippet.toByteArray())
+            sourceFile
+        }
+
+        // TODO: make sure that the source file is copied into a source directory and the right dir structure
+        myFixture.configureFromExistingVirtualFile(sourceFile)
+        val psiFile = myFixture.file
+        assertEquals(psiFile.fileType, KotlinFileType.INSTANCE)
+
+        // TODO: fix compileCodeSnippet and copy data properly into build folder
+        val classOutputDir = compileCodeSnippet(codeSnippet)!!
+        val target = myFixture.copyDirectoryToProject(classOutputDir.path, buildFolder.path)
+        assertNotNull(target)
```

Comment:
This assert could be in the testing library.

Reply:
We're not building assertions in the testing dsl